### ISSUE TITLE
POST handler for adding a notebook to the running instance

### ISSUE THREAD

**Issue by parente** [jupyter-incubator/dashboards_nodejs_app#15]
It would be handy if the node app supported `POST /notebooks` to add / overwrite an existing notebook with the same with a new one for quick and dirty deployment.

https://github.com/jupyter-incubator/dashboards_nodejs_app/blob/master/routes/index.js

Let's not worry yet about persisting these or PUT vs POST or overwrite vs error. I want to get a simple demonstration of how a bundler can be written for Jupyter Notebook server to post a simple notebook to this API to get it to show up as a dashboard instantly.

**Comment by dalogsdon** [comment]
Discussed with @jtyberg. We'll go with a `multipart/form-data` handler to upload a file. It will upload to the `NOTEBOOKS_DIR` directory and overwrite any existing file.

The signature will follow the `GET /notebooks/*` API, so `POST /notebooks/*` will upload a file with optional subdirectory. Example: `POST /notebooks` will upload to the root data dir while `POST /notebooks/foobar` will create (if necessary) and upload to a folder called "foobar" in the data dir.

I have the initial implementation working. I still need to clean up a bit and finish a bit of error handling.

**Comment by dalogsdon** [comment]
The form data key to use for the file is "file".

Example of calling from python (based on [this example](http://docs.python-requests.org/en/latest/user/quickstart/#post-a-multipart-encoded-file)):

```python
import requests
url = 'http://192.168.99.100/notebooks/newdashboard'
file = { 'file': open('my_dashboard.ipynb', 'rb') }
requests.post(url, files=file)
```

The uploaded dashboard filename will be based on the URL, not on the original file name. This way the same POST URL can be used in the browser to view the dashboard. In the python example above, the newly uploaded notebook can be viewed at `http://192.168.99.100/notebooks/newdashboard`

**Comment by parente** [comment]
PR uses multer, so for future reference when we don't want to just dump the nb to local disk: https://github.com/expressjs/multer/blob/master/StorageEngine.md

**Comment by parente** [comment]
Closed by PR #43.
